Download/Export Presentation with Annotations

danielpetri1 opened the pull request:
### What does this PR do?
Adds two new messages to BBB's core to enable the following:

- Allow attendees to download presentation with annotations during the meeting
- Export presentation with annotations from breakout rooms when breakout ends to main meeting

The PDF is generated by a Node application in the `export-annotations` directory. When running, it'll take the requests made by users from Redis containing the state of the whiteboard at the time of the request. It's intended as a service running in the background at all times, but under some use cases it may be sensible to only start the app when desired, for instance after the meeting has ended to recreate the whiteboard at specific timestamps.

### **ExportPresentationWithAnnotationReqMsg**

This message generates a PDF file that is uploaded into the main room, in order to bring back content from breakout rooms.
In the short video below, pressing the button sends out the message containing the following attributes:

- parentMeetingId (set to the meetingId for demonstration purposes)
- allPages: true | false

When allPages is false, only the current presentation page is exported. The resulting file is automatically listed under the available presentations.

https://user-images.githubusercontent.com/33319791/156412356-f0269e38-88b1-437d-bb4c-5beb8a1a7cb3.mov

### **MakePresentationWithAnnotationDownloadReqMsg**
- presId
- allPages: true | false
- pages: [] - list of pages when allPages is false, e.g. [1, 3, 8]

When sent out, generates a PDF with the desired annotated slides and replies with a link to the file in the MeetingActor. Can therefore also be used to take a snapshot of the current slide. In the video below, I paste the resulting URL a browser to acess the PDF.

https://user-images.githubusercontent.com/33319791/156415260-62209a22-2c62-4b37-b640-51124757b216.mp4

### Architecture:
![Architecture](https://user-images.githubusercontent.com/33319791/157302750-f1ec2865-9da6-4dcf-9fc6-394162f024ba.png)

### Requires:
rsvg-convert (`sudo apt-get install librsvg2-bin`)
wkhtmltoimage ([installation steps](https://websiteforstudents.com/how-to-install-wkhtmltopdf-wkhtmltoimage-on-ubuntu-18-04-16-04/))

### Usage:
To start the NodeJS app, run `npm install` in the `export-annotations` folder followed by `npm start`. To generate the PDF link, set the host in `export-annotations/config/settings.json` and make sure the change from `presentation-slides.nginx` is applied.

### Replaces:
PR [#13666] - Bring content back from breakout rooms to main room for presenter

PR [#13016] - Post-publish presentation script to export PDF with annotations

### Additional information:
This PR does not implement any client-side messages, meaning that this functionality is not exposed in the client. This PR can be treated as an API for the client to make use of the feature.

frankemax committed:
feat(click-outside): close emoji picker on outbound clicks

add the click-outside component and modifies the message-form/text-input
components to apply the new feature

(cherry picked from commit 654357e6984160fe8ea074fcfa1103f7b51500bc)

antobinary commented on labs/vertx-akka/src/main/scala/org/bigbluebutton/client/meeting/AllowedMessageNames.scala:
```diff
@@ -61,6 +61,10 @@ object AllowedMessageNames {
     CreateNewPresentationPodPubMsg.NAME,
     RemovePresentationPodPubMsg.NAME,
     SetPresenterInPodReqMsg.NAME,
+    MakePresentationWithAnnotationDownloadReqMsg.NAME,
```
Wow, you even updated `vertx-akka`, wow!

danielpetri1 commented:
Hi @antobinary,

In our previous call with the spontaneous demonstration I don't think I pointed out that if this app is **not** running as a service in the background, it will still generate all the PDFs once it is fired up even if the meeting is long gone.

So the use case that you mentioned where there is a snapshot of the whiteboard being taken once every X minutes is possible with this, the only change would be to stop it from running as soon as no more jobs are in the Redis queue instead of leaving it listening all the time.

TiagoJacobs committed:
Add github workflow to run automated tests

prlanzarin committed:
refactor: use webrtc-adapter package, remove static script from compatibility

TiagoJacobs committed:
Filter branches that CI tests will run (>=2.5.x)

antobinary committed:
build(fsesl-akka): build deps in sequence to allow for ci use

lfzawacki committed:
feat(layout): Both presenters and moderators can set pushLayout

ramonlsouza committed:
primary circle button - revised  styles

imdt-joaov committed:
refactor: sidebar content panel UI polishing

lfzawacki committed:
fix(layout): Moderators could not change layout if pushLayout was true

gabriellpr committed:
removing white spaces inside menus

refactoring code

...

frankemax committed:
feat(message-form): add emoji button

germanocaumo committed:
fix viewer crash when there is no presentation + some cleanup

GuiLeme committed:
Updated nokogiri to version 1.13.5

antobinary committed:
Merge pull request #15049 from GuiLeme/update-nokogiri-1.13.5

chore: Updated nokogiri to version 1.13.5

KDSBrowne committed:
remove orphan console log

imdt-joaov committed:
improvement: chat and user-list export file name date string

frankemax committed:
feat(emoji-picker): exclude some emojis

exclude emojis not supported by chromium-based browsers

(cherry picked from commit c1a904b842438f1d70949921133e4b89cdcae651)

imdt-joaov committed:
Merge remote-tracking branch 'upstream/v2.5.x-release' into date-zero-padding-chat-userlist

antobinary committed:
fix: set target to jvm-1.11 in akka-bbb-fsesl

germanocaumo committed:
fix persist of other tldraw shape operations (including groups and copy/paste) +
fix shape not persisting when other user than the owner changed it

germanocaumo committed:
fix typo

antobinary committed:
Merge pull request #15054 from antobinary/ci-25

ci: Add github workflow to run automated tests (was #14880)

prlanzarin committed:
Merge pull request #14881 from prlanzarin/u26-blind-moss2

refactor: use webrtc-adapter package, remove static script from compatibility

ramonlsouza committed:
Merge pull request #14994 from KDSBrowne/2.6-bbb-tldraw

feat: Integrate TLDraw with BigBlueButton (part 1)

ramonlsouza committed:
Merge pull request #15050 from JoVictorNunes/date-zero-padding-chat-userlist

improvement: chat and user-list export file name date string

ramonlsouza committed:
Merge pull request #15048 from gabriellpr/ui-fix-align

style: Removing and alinging dropdowns items.

ramonlsouza committed:
Merge pull request #15013 from ramonlsouza/multi-tab-fix

fix: prevent user join from multiple tabs with same sessionToken

ramonlsouza committed:
increase dropdown menu padding

ramonlsouza committed:
Merge pull request #15062 from ramonlsouza/increase-menu-padding

style: increase dropdown menu padding

ramonlsouza committed:
increase dropdown menu padding

Tainan404 committed:
Refactor aria alert and remove inner-htmls from the client

gabriellpr committed:
adding border inside shared notes for viewers

ramonlsouza committed:
Merge pull request #15066 from gabriellpr/border-on-pads

Adding border inside shared notes for viewers

ramonlsouza committed:
Merge pull request #15064 from ramonlsouza/increase-menu-padding-

style: increase dropdown menu padding

danielpetri1 committed:
Merge branch 'v2.6.x-release' into tldraw-with-annotation-export

danielpetri1 committed:
Add temporaryPresentationId parameter for upload

Arthurk12 committed:
build(webhooks): v2.6.0

Update 'bbb-webhooks' version.

https://github.com/bigbluebutton/bbb-webhooks/releases/tag/v2.6.0

danielpetri1 committed:
Sticky note support

Tainan404 committed:
Make sure that user is a chat participant when send a message

frankemax committed:
feat(video-avatar): hide icons when squeezed

ramonlsouza committed:
Merge pull request #15044 from JoVictorNunes/issue-15030

refactor: sidebar content panel UI polishing

Commit:
Fix: voice users not removed from participants list

`removeUser` was called with wrong parameters

imdt-joaov committed:
fix: point media button chevrons upwards #15068

Tainan404 committed:
Merge remote-tracking branch 'upstream/v2.5.x-release' into remove-innerHTML

ramonlsouza committed:
Merge pull request #15067 from Tainan404/remove-innerHTML

refactor: Remove inner html

ramonlsouza committed:
Merge pull request #15081 from JoVictorNunes/issue-15068

fix: point media button chevrons upwards

danielpetri1 committed:
Implement drraw tool

imdt-joaov committed:
fix(camera): prevent dock position change on click

gabriellpr committed:
adding url styles

...

styling external links inside the banner

Refactoring code

adding underline on hover and focus

removing hasLiink

KDSBrowne committed:
move multi user WB button to presentation toolbar (100% width) | reduce zoom increment

prlanzarin committed:
build: bump bbb-webrtc-sfu to v2.8.2

prlanzarin committed:
Merge pull request #15086 from bigbluebutton/sfu282

build: bump bbb-webrtc-sfu to v2.8.2

ramonlsouza committed:
Merge pull request #15082 from JoVictorNunes/issue-15036

fix(camera): prevent dock position change on click

Tainan404 committed:
Make sure do not show wrong data in user left message

gustavotrott committed:
Merge pull request #15076 from Tainan404/check-participants

ramonlsouza committed:
Merge pull request #15087 from Tainan404/fix-userleft

Make sure do not show wrong data in user left message

ramonlsouza committed:
Merge pull request #15084 from gabriellpr/external-links-style

style: Adding revised styles for our external links

frankemax committed:
fix(message-form): send message with emoji

gustavotrott committed:
Merge pull request #15053 from bigbluebutton/antobinary-patch-1

imdt-joaov committed:
fix: ghost button focus shadow

ramonlsouza committed:
Merge pull request #15089 from JoVictorNunes/fix-focus-shadow

fix: ghost button focus shadow

ramonlsouza committed:
Merge pull request #15080 from schrd/issue-14969

Fix: voice users not removed from participants list

frankemax committed:
feat(video-avatar): change border

Apply isTalking border with a pseudo-element.

prlanzarin committed:
fix(audio): remove deprecated getLocalStreams usage

Use the built-in getLocalStream from the peer wrapper instead (which
relies on getSenders - the proper, spec-compliant way).

Two different issues are addressed:
  - RTCPeerConnection.getLocalStreams is a pre-1.0 WebRTC spec which is
    deprecated and not recommended.
  - Fixed an issue where a switch from full audio to listen only could
  cause the latter to be rejected with an error 1004 in rare scenarios.

prlanzarin committed:
Merge pull request #15093 from prlanzarin/u26-ascendant-ecrit

fix(audio): remove deprecated getLocalStreams usage

lfzawacki committed:
feature(layout): Bring back layoutType propagation, but disallow 'custom' layout in mobile screens

KDSBrowne committed:
disable pan when multiuser access is off

KDSBrowne committed:
stop rendering curors when they leave the WB area

imdt-joaov committed:
fix(chat notification): duplication of first message notification

KDSBrowne committed:
allow pan when multiuser enabled

frankemax committed:
fix(video-avatar): prevent state update on an unmounted component

bbb-localization-server committed:
chore: Pulled the latest 2.4 HTML5 locales from Transifex

GuiLeme committed:
[update-ruby-gems] - Update of nokogiri and rack-gem

antobinary committed:
Merge pull request #15105 from GuiLeme/update-nokogiri

chore(recording): Updated Ruby Gems (nokogiri, rack-gem)

ramonlsouza committed:
Merge pull request #15104 from ramonlsouza/bbb25-locales-may30

chore: Pulled the latest 2.5 HTML5 locales from Transifex

frankemax committed:
fix(webcam): user status icons align

Align user status icons to match other icons paddings/margins

imdt-joaov committed:
Rename custom hook

imdt-joaov committed:
chore(Dashboard): run npm audit fix

antobinary committed:
Merge pull request #15108 from antobinary/may-30-ci

build(fsesl-akka): build deps in sequence to allow for ci use

antobinary committed:
Merge pull request #15107 from JoVictorNunes/chore-npm-audit-May-30

chore(Dashboard): run npm audit fix

frankemax committed:
feat(message-form): send emoji at caret position

danielpetri1 committed:
Draw: support dashed and dotted types

ramonlsouza committed:
restore removed styles

mariogasparoni committed:
Merge pull request #15075 from frankemax/emoji-picker-v3

feat(emoji-picker): add component

imdt-joaov committed:
revert: unifying of click handlers of the default backgrounds

imdt-joaov committed:
revert: unifying of click handlers of custom backgrounds

imdt-joaov committed:
fix: localize default button label for custom backgrounds

ramonlsouza committed:
undo double quotes replace

ramonlsouza committed:
Merge pull request #15103 from KDSBrowne/2.6-bbb-tldraw

feat(whiteboard): Move Multiuser Whiteboard Button to Presentation Toolbar

frankemax committed:
fix(webcam): draggable camera dock

danielpetri1 committed:
Rectangles: support fill

imdt-joaov committed:
fix: add some file sanitization

germanocaumo committed:
feat(whiteboard): poll result, zoom fit slide +

- Poll result as text shape in the lower right-hand
- Include all tldraw shape bounds as size in akka (to help Daniel rendering in pdf export)
- Default/initital zoom will be centered in the slide and maximize the size according to the presentation area
- Also limited the max zoom out to be the one that fits the slide, to be similar as we had before

imdt-joaov committed:
fix: actions bar button styling

mariogasparoni committed:
Merge pull request #15106 from frankemax/camera-avatar-polishment

feat(webcam) camera avatar polishment

kepstin committed:
Fix rap-enqueue.rb to load gems with bundler

In BBB 2.5, we switched the recording system to use bundled gems
included privately in the recording package, rather than installed
system-wide. The rap-enqueue.rb script needs to be updated to load the
bundler gems.

According to bundler devs, setting the BUNDLE_GEMFILE environement
variable is the supported way to tell bundler where to find it
(otherwise bundler will search starting at the current working directory
- which in the case of rap-enqueue.rb is probably nowhere near the
Gemfile).

Use a relative path from the directory where the script is located so it
can be run both when installed and from a development environment.

Switch the script interpreter to use /usr/bin/env to load ruby from the
path. Doesn't make a difference in the installed package, but it makes
testing on development systems with multiple ruby environments easier.

Fixes #15085

danielpetri1 committed:
Initial ellipse support

danielpetri1 committed:
Basic triangle support

gabriellpr committed:
export chat save and copy updated

adding system messages key to public_chat_clear

danielpetri1 committed:
Merge remote-tracking branch 'upstream/v2.6.x-release' into tldraw-with-annotation-export

imdt-joaov committed:
improvement: add HOC to concentrate drag-and-drop code

germanocaumo committed:
fix(whiteboard): missing import

ramonlsouza committed:
Merge pull request #15115 from JoVictorNunes/fix-button-style-May-31

fix: actions bar button styling

imdt-joaov committed:
improvement: add a new service function for reading the file

imdt-joaov committed:
fix: state update on delete

ramonlsouza committed:
Merge pull request #15114 from germanocaumo/tldraw-improvs

feat(whiteboard): poll result, zoom fit slide +

imdt-joaov committed:
fix: move config key to public.virtualBackgrounds (Settings.yml)

imdt-joaov committed:
fix: make drag-and-drop stuff follow the config

imdt-joaov committed:
fix: remove style prop 'isVisualEffects' from button props

antobinary committed:
Merge pull request #15074 from Arthurk12/webhooks-update

build(webhooks): v2.6.0

imdt-joaov committed:
Merge remote-tracking branch v2.6.x-release into virtual-background-upload

imdt-joaov committed:
Remove merge residue

ramonlsouza committed:
Merge pull request #14794 from JoVictorNunes/virtual-background-upload

feat(video): upload of own webcam virtual background

danielpetri1 committed:
Use probe-image-size for dimensions; embed SVGs in place of PNGs

ramonlsouza committed:
Merge pull request #15095 from ramonlsouza/issue-7653

style: Revised styles for our primary circle buttons

KDSBrowne committed:
fix cursors lock for viewers

ramonlsouza committed:
do not group moderator message sent to all breakouts

ramonlsouza committed:
Merge pull request #15078 from lfzawacki/develop-layout

fix(layout): Fix propagation bugs and allow moderators to set the pushLayout flag

imdt-joaov committed:
fix(chat): show new message indicators for poll mesages

antobinary committed:
Merge pull request #15116 from kepstin/issue-15085

Fix rap-enqueue.rb to load gems with bundler

imdt-joaov committed:
fix(Dashboard): make username look more clickable

gabriellpr committed:
fix: pad line space not showing

flex column removed

ramonlsouza committed:
Merge pull request #15128 from gabriellpr/line-space-pads

fix: pad line space not showing for viewers

gabriellpr committed:
Swap version of bbb-diff npm dependency

ramonlsouza committed:
audio modal spacing polishing

ramonlsouza committed:
Merge pull request #15121 from KDSBrowne/2.6-lock-cursor

Restore Viewer Cursors Lock Functionality

ramonlsouza committed:
revised styles primary button

ramonlsouza committed:
adjust external video button

ramonlsouza committed:
remove unused custom button styles

KDSBrowne committed:
fix zoom value reset on unmount / mount

danielpetri1 committed:
Merge branch 'v2.6.x-release' into tldraw-with-annotation-export

danielpetri1 committed:
Text alignment & rotation

ramonlsouza committed:
Update bigbluebutton-html5/imports/ui/components/common/button/styles.js

Co-authored-by: João Victor Nunes <62393923+JoVictorNunes@users.noreply.github.com>

ramonlsouza committed:
Merge pull request #15134 from ramonlsouza/issue-7744

style: Revised styles for our primary button

imdt-joaov committed:
fix(tooltip): wait 150ms to hide the tooltip

danielpetri1 committed:
Line support

antonbsa committed:
test: update playwright to v1.19.2

antonbsa committed:
test: fix/update tests on v2.5

ramonlsouza committed:
re-sub to current poll when presenter changes

antobinary committed:
chore: Bump version to 2.5.0

danielpetri1 committed:
Curved line and arrow support

ramonlsouza committed:
Merge pull request #15131 from ramonlsouza/audio-modal-polishing

Audio modal polishing

ramonlsouza committed:
remove unused code

antonbsa committed:
ci: improve automated tests workflow

imdt-joaov committed:
chore: fix Learning A. Dashboard npm dependencies

GuiLeme committed:
fix user removal in breakout

ramonlsouza committed:
rename snapshot button label

gustavotrott committed:
Merge pull request #15147 from GuiLeme/fix-user-removal-breakout

fix user removal in breakout

danielpetri1 committed:
Take child index into account

frankemax committed:
feat(audio): unify audio buttons

Move device selection dropdown to microphone button
Move option 'leave audio' to an option within the dropdown
Remove the audio exit button when device !== mobile

imdt-joaov committed:
fix(connection status modal): network status icon color #15133

imdt-joaov committed:
fix(connection status modal): data saving settings button

ramonlsouza committed:
fix user not removed from users context on leave

ramonlsouza committed:
Merge pull request #15155 from ramonlsouza/issue-15069

fix: user not removed from users context on leave

gabriellpr committed:
etherpad cutting text

fixing etherpad alignment for lock viewers

removing break words

fix: break-word

adding white-space

ramonlsouza committed:
Revert "fix: user not removed from users context on leave"

ramonlsouza committed:
Merge pull request #15164 from bigbluebutton/revert-15155-issue-15069

Revert "fix: user not removed from users context on leave"

antobinary committed:
chore: Bump version to 2.5.1

antobinary committed:
Merge pull request #15146 from antonbsa/2.5-tests-fix-and-ci-improvements

test: v2.5 tests fixes and CI improvements

KDSBrowne committed:
disable dropping of assets on whiteboard

KDSBrowne committed:
recover presentation option menu | position inside tldraw menu

GuiLeme committed:
[issue-15051] - Refactored all traces from Red5

imdt-joaov committed:
fix: presenter indicator in RTL layout #15168

KDSBrowne committed:
stop zoom out at min %

KDSBrowne committed:
Merge branch '2.6-toolbar-zoom-val' of https://github.com/KDSBrowne/bigbluebutton into 2.6-toolbar-zoom-val

danielpetri1 committed:
Circle: use solid with instead of draw

danielpetri1 committed:
Improve arrow bend and head

mariogasparoni committed:
fix(audio): moderator not able to transfer audio to breakout room

Closes #15169

danielpetri1 committed:
Add shape labeling feature

ramonlsouza committed:
Merge pull request #15117 from gabriellpr/export-chat-bug

fix: missing "chat has been cleared" message at chat export

ramonlsouza committed:
Merge pull request #15130 from gabriellpr/replace-dependency

chore: Swap version of bbb-diff npm dependency

ramonlsouza committed:
Merge pull request #15124 from JoVictorNunes/issue-15113

fix(chat): show new message indicators for poll mesages

ramonlsouza committed:
Merge pull request #15123 from ramonlsouza/issue-15112

fix: Wrong aggregation of broadcast messages to breakout room

ramonlsouza committed:
Merge pull request #15148 from ramonlsouza/rename-snapshot-feature

Rename snapshot button label

ramonlsouza committed:
Merge pull request #15153 from JoVictorNunes/issue-15133

fix(connection status modal): network status icon

ramonlsouza committed:
Merge pull request #15173 from JoVictorNunes/issue-15168

fix: presenter indicator in RTL layout

ramonlsouza committed:
Merge pull request #15139 from JoVictorNunes/issue-15060

fix(tooltip): wait 150ms to hide the tooltip

ramonlsouza committed:
Merge pull request #15102 from JoVictorNunes/v2.5.x-release

fix(chat notification): duplication of first message notification

danielpetri1 committed:
Remove twmoji, increase wkhtmltoimage security, sticky tool positioning

ramonlsouza committed:
Merge pull request #15145 from JoVictorNunes/dashboard-chore-Jun.08

chore: fix/update Learning A. Dashboard npm dependencies

ramonlsouza committed:
Merge pull request #15125 from JoVictorNunes/issue-15122

fix(Dashboard): make username look more clickable

ramonlsouza committed:
Merge pull request #15143 from ramonlsouza/current-poll-presenter

fix: re-subscribe to current poll when presenter changes

gustavotrott committed:
Merge pull request #15176 from mariogasparoni/fix-15169

antobinary committed:
fix: add missing emoji-mart in html5 package-lock.json

antobinary committed:
Merge pull request #15182 from antobinary/npm-html5

fix: add missing emoji-mart in html5 package-lock.json

antobinary committed:
chore: Bump version to 2.5.2

ramonlsouza committed:
Merge pull request #15156 from gabriellpr/etherpad-cutting-text

fix: Etherpad cutting off text

danielpetri1 committed:
Refactor Scala: immutability, no returns, recursion instead of loops

antobinary committed:
chore: set 2.5 as the recommended version, discontinue 2.3

frankemax committed:
fix(audio): select device dropdown colors

Set list titles colors to colorGrayDark
Set active devices background color to off-white

frankemax committed:
fix(audio): select device dropdown icon position

Move the 'leave' icon to the left-hand side

antobinary committed:
Merge pull request #15187 from bigbluebutton/antobinary-patch-1

docs: set 2.5 as the recommended version, discontinue 2.3

danielpetri1 committed:
Add shape fill for pencil drawings

KDSBrowne committed:
add comment

KDSBrowne committed:
replace zoom minBound with zoomed state

GuiLeme committed:
[issue-15160] - fixed error logs in recordings

ramonlsouza committed:
Merge pull request #15154 from frankemax/unify-audio-buttons-27

feat(audio): unify audio buttons

ramonlsouza committed:
Merge tag 'v2.5.1' into merge25-26-jun16

antobinary committed:
Merge pull request #15201 from ramonlsouza/merge25-26-jun16

chore: Merge v2.5.1 into v2.6

antobinary committed:
Merge pull request #15195 from GuiLeme/issue-15160

(fix) - Fixed error logs in recordings about etherpad

KDSBrowne committed:
fix viewer zoom reset when presenter remounts presentation

KDSBrowne committed:
restore WB snapshot image download

ramonlsouza committed:
Merge pull request #15137 from KDSBrowne/2.6-toolbar-zoom-val

Display Correct Zoom Value on Presentation Restore

ramonlsouza committed:
Merge pull request #15189 from KDSBrowne/2.6-disable-fd

fix: Disable File Drop on Whiteboard

GuiLeme committed:
Merge remote-tracking branch 'upstream/v2.6.x-release' into issue-15051

danielpetri1 committed:
Refactor filled shapes

ramonlsouza committed:
update bbb-icons

bbb-localization-server committed:
chore: Pulled the latest 2.5 HTML5 locales from Transifex

antobinary committed:
Merge pull request #15225 from antobinary/loca

chore: Pulled the latest 2.5 HTML5 locales from Transifex

KDSBrowne committed:
fix menu position in rtl and iPhone

KDSBrowne committed:
Merge branch 'v2.6.x-release' into 2.6-wb-options-dropdown

antobinary committed:
chore: Merged BBB 2.5.2 into v2.6.x-release

ramonlsouza committed:
Merge pull request #15190 from KDSBrowne/2.6-wb-options-dropdown

fix: Recover presentation options menu

danielpetri1 committed:
Use ImageMagick+Pango for text rendering

ramonlsouza committed:
Merge pull request #15232 from antobinary/merge-252

chore: Merge BBB 2.5.2 into 2.6.x

prlanzarin committed:
fix(audio): remove userLeftFlag from global audio permission check

If the user is in akka-apps' user cache (which means they are still
visible in the user list) and triggers an audio re-connect, the
re-connect should be allowed.

If the user indeeds fails to be reconnected to the whole system, then
the audio session will be ejected when the user is completely ejected
from the system.

prlanzarin committed:
fix(audio): decouple mic auth RPCs from global audio (SFU)

The SFU microphone module was using global audio authentication RPCs in
akka-apps to determine whether a mic connection is valid.

Fixes an issue where SFU microphone connections wouldn't take lock
settings in account in the permission checks.

In detail: decouple them by adding GetMicrophonePermissionReqMsg and
RespMsg. Allows flexibility on which conditionals to use in both. IMO it
is also better than renaming+adding a boolean to GetGlobalAudioPermission*
due to the fact that it gets easier to guarantee backwards compatibility
between SFU and BBB (ie SFU 2.9 keeps working on 2.4, 2.5).

prlanzarin committed:
fix(webcam): remove userLeftFlag from permission checks

If the user is in akka-apps' user cache (which means they are still
visible in the user list) and triggers a camera re-connect, the
re-connect should be allowed to minimize outages.

If the user indeeds fails to be reconnected to the whole system, then
the camera sessions will be ejected when the user is completely ejected
from the system.

prlanzarin committed:
fix(screenshare): remove userLeftFlag from permission checks

If the user is in akka-apps' user cache (which means they are still
visible in the user list) and triggers a screenshare re-connect, the
re-connect should be allowed to minimize outages.

If the user indeeds fails to be reconnected to the whole system, then
the screen sharing sessions will be ejected when the user is
completely ejected from the system.

antobinary committed:
fix: tweak transifex.sh to the new 2.6 Transifex project

antobinary committed:
fix: tweak transifex.sh to the new 2.6 Transifex project

antobinary committed:
Merge pull request #15242 from antobinary/merge-252

fix: tweak transifex.sh to the new 2.6 Transifex project

bbb-localization-server committed:
chore: Pulled the latest 2.6 HTML5 locales from Transifex

antobinary committed:
Merge pull request #15243 from antobinary/merge-252

chore: Pulled the latest 2.6 HTML5 locales from Transifex

ramonlsouza committed:
add big recording indicator for viewers when record is active

KDSBrowne committed:
disable hover highlight for background slide image

prlanzarin committed:
Merge pull request #15234 from prlanzarin/u26/fix/ghi15233

fix: remove userLeftFlag from audio, camera and screen sharing auth checks

prlanzarin committed:
build(bbb-webrtc-sfu): v2.9.0-alpha.2

See https://github.com/bigbluebutton/bbb-webrtc-sfu/releases/tag/v2.9.0-alpha.2

GuiLeme committed:
[issue-14852] - Change in bbb-conf to not rely on tomcat

prlanzarin committed:
Merge pull request #15264 from bigbluebutton/sfu290a2

build(bbb-webrtc-sfu): v2.9.0-alpha.2

GuiLeme committed:
[issue-14852] - Corrected some debug leftovers.

danielpetri1 committed:
Workaround CairoSVG issue rendering blackboxes in transparent text

antobinary committed:
Merge pull request #15274 from antobinary/cherry-pick-15265

fix(bbb-conf): Change in bbb-conf to not rely on tomcat (port of #15265)

KDSBrowne committed:
disable shape selection for users without multiuser access

ramonlsouza committed:
prevent chat-context crash

antobinary committed:
Merge pull request #15277 from ramonlsouza/fix-chat-context-adapter-check

fix: prevent chat-context crash on missing data

imdt-joaov committed:
fix: add check for potential undefined userId

antobinary committed:
Merge pull request #15280 from JoVictorNunes/undefined-values-Jun-28

fix: add check for potential undefined userId

danielpetri1 committed:
BG slide quality control, cropped PDF support

ramonlsouza committed:
Merge pull request #15223 from ramonlsouza/bbb-icons-jun21

chore: pull latest bbb-icons, add woff2 font

ramonlsouza committed:
Merge pull request #15278 from KDSBrowne/2.6-disable-hover-slidebg

Disable Background Slide Image Hover Highlight

antobinary committed:
chore: cleanup of vertx-akka

antobinary committed:
Merge pull request #15288 from antobinary/cleanup2

chore: cleanup of vertx-akka

antobinary committed:
chore: remove .travis.yml

antobinary committed:
chore: remove obsolete files

antobinary committed:
chore: remove obsolete doc/message.md

antobinary committed:
Merge pull request #15289 from antobinary/cleanup2

chore: cleanup of Travis + a few random files

ramonlsouza committed:
Merge pull request #15256 from ramonlsouza/issue-15252

fix: Recording indicator too subtle

antobinary committed:
Merge pull request #15171 from GuiLeme/issue-15051

refactor: Removed all traces of Red5

danielpetri1 committed:
Merge remote-tracking branch 'upstream/v2.6.x-release' into container-approach
